### PR TITLE
fix(core):  switch, fix a11y issues

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue.
Closes SAP/fundamental-ngx/#6184

## Description
Fixed switch a11y issues

##### During Implementation
1. Visual Testing:
- [n/a] visual misalignments/updates
- [n/a] check Light/Dark/HCB/HCW themes
- [n/a] RTL/LTR - proper rendering and labeling
- [n/a] responsiveness(resize)
- [n/a] Content Density (Cozy/Compact/(Condensed))
- [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [X] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [n/a] Mouse vs. Keyboard support
- [n/a] Text Truncation
2. API and functional correctness
- [n/a] check for console logs (warnings, errors)
- [n/a] API boundary values
- [n/a] different combinations of components - free style
- [n/a] change the API values during testing
3. Documentation and Example validations
- [n/a] missing API documentation or it is not understandable
- [n/a] poor examples
- [X] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [X] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [n/a] Run npm run build-pack-library and test in external application
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review
- [x] Mouse vs. Keyboard support
2. API and functional correctness
n/a
3. Documentation and Example validations
n/a
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/switch/examples/switch-binding-example/switch-binding-example.component.scss [hunk 1]
```diff
@@ -10,3 +10,7 @@ button {
 fd-switch {
     margin-bottom: 20px;
 }
+
+.switch-info {
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/switch/examples/switch-binding-example/switch-binding-example.component.scss:14]: should be prefixed with `docs` so the user is aware this is a class for the documentation.

Comment (InnaAtanasova) [reply apps/docs/src/app/core/component-docs/switch/examples/switch-binding-example/switch-binding-example.component.scss:14]: I'd better apply inline style for the margin :)

Comment (dmitry-stepanenko) [reply apps/docs/src/app/core/component-docs/switch/examples/switch-binding-example/switch-binding-example.component.scss]: done

Comment (Betrozov) [reply apps/docs/src/app/core/component-docs/switch/examples/switch-binding-example/switch-binding-example.component.scss]: 👍

**Review by InnaAtanasova — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
🚀

**Review by N1XUS — APPROVED** [review]

File: libs/core/src/lib/switch/switch.component.html [hunk 2]
```diff
@@ -7,8 +7,8 @@
     [class.is-disabled]="disabled"
     [attr.aria-label]="ariaLabel"
 >
-    <span *ngIf="activeText || inactiveText" class="fd-switch__text">
-        {{ isChecked ? activeText : inactiveText }}
+    <span *ngIf="(isChecked ? activeText : inactiveText) as text" class="fd-switch__text">
```
Comment (Smiranin) [inline libs/core/src/lib/switch/switch.component.html:10]: I am a little confused by this check, why should we add `as text` expression?
May be this checking would be more understandable `!!(isChecked ? activeText : inactiveText)`

Comment (dmitry-stepanenko) [reply libs/core/src/lib/switch/switch.component.html:10]: @Smiranin I'm just avoiding keeping the same condition in 2 places: in the `ngIf` and below as an interpolation.

If you want, I can make a getter on the component level instead. But in my opinion it's better to not pollute component with extra properties without a need.

Comment (Smiranin) [reply libs/core/src/lib/switch/switch.component.html:10]: Ok. No need

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by puru-hk** [comment]
Issue1: Not fixed 
Observed: On activating and deactivating the ‘Switch 1’ and Switch 2’ button, when ‘Switch 1’ is activated the status of ‘Switch 1’ and ‘Switch 2’ status is read.

Expected: On activating/Deactivating the ‘Switch 1’ and ‘Switch 2’ button, status of the selected switch button status should be read

Screenshot 2021-09-23 at 17 28 37

Issue 2: fixed 

Issue3 : Not fixed

Observed: Associated label is not read

Expected: On accessing the Switch its associated label should be read
Screenshot 2021-09-23 at 17 28 51

**Comment by dmitry-stepanenko** [comment]
@puru-hk 
fixed issue 3

issue 1 appears to be fine, I've tested it both on MacOS with VoiceOver and on Windows 10 with JAWS
JAWS (it's a bit in russian, but you can see the overall output is correct):
image
VoiceOver
https://user-images.githubusercontent.com/33101123/134661193-0fa72dd0-e456-49bc-a577-aac3d439f96a.mp4

**Comment by puru-hk** [comment]
@dmitry-stepanenko Looks fine

**Pull request merged**